### ISSUE TITLE
FP on let's encrypt validation agent

### ISSUE THREAD

**Issue by theMiddleBlue** [coreruleset/coreruleset#3163]
932200 prevent ACME challenge from the letsencrypt agent

```shell
$ curl -s -H 'x-crs-paranoia-level: 2' -H "x-format-output: txt-matched-rules" -H "User-Agent: Mozilla/5.0 (compatible; Let's Encrypt validation server; +https://www.letsencrypt.org)" "https://sandbox.coreruleset.org/"

932200 PL2 RCE Bypass Technique
949110 PL1 Inbound Anomaly Score Exceeded (Total Score: 5)
980170 PL1 Anomaly Scores: (Inbound Scores: blocking=5, detection=5, per_pl=0-5-0-0, threshold=5) - (Outbound Scores: blocking=0, detection=0, per_pl=0-0-0-0, threshold=4) - (SQLI=0, XSS=0, RFI=0, LFI=0, RCE=5, PHPI=0, HTTP=0, SESS=0, COMBINED_SCORE=5)
```

**Comment by theMiddleBlue** [comment]
we should really remove ua and referer from this kind of rules

**Comment by franbuehler** [comment]
Thank you for testing these rules 🙏 

It seems that the bypass list regarding the request headers cannot be solved so easily. Would have been too easy ;)
We'll have to decide whether we want to check the two request headers at a higher PL or not at all.

Do you have a list of the problematic rules?

Commit that introduced this new FP: 
https://github.com/coreruleset/coreruleset/commit/82d427ede1479204be93c1a36b9d856240732347

**Comment by theMiddleBlue** [comment]
Thank you @franbuehler 

unfortunately I'm still collecting data, but I can say that SQLi and RCE ruleset have increased the number of FPs on 4.0. List and Bypass prevention rules are the most problematic AFAICS (always because UA and referer request headers). 

I think also that we miss an engine functionality for inspecting Referer header. Let me clarify (Sorry for using this space to talk about this. If anyone is interested, we can discuss it further in the chat or in a separate issue.):

The referer request header is really different from other headers, it should contains a URL with schema, domain, path, and querystring (depending on the referer policy applied by the browser). To allow CRS to correctly inspect this kind of header, I think we should have something on the engine that parses it, in order to have `REFERER_ARGS` or `REFERER_REQUEST_URI` variables (I really don't know if this is possible as is). 

Otherwise, inspecting a full URL with rules developed for ARGS will always produce FPs I guess.

my 2c

**Comment by dune73** [comment]
Good thinking @theMiddleBlue.

**Comment by azurit** [comment]
I believe we are able to do it by our own using only SecLang.

**Comment by dune73** [comment]
Could be worth a try. I mean there is this idea where we add TX:/^crs_args_/ to all rules that inspect ARGS (in time for CRS4). This could use this functionality.

**Comment by azurit** [comment]
Here is an example rule which diggs up `REQUEST_URI` from the `Referer`:
```
SecRule REQUEST_HEADERS:Referer "@rx ^https?://.*?(/.*)" \
    "id:40,\
    phase:1,\
    pass,\
    capture,\
    t:none,\
    nolog,\
    setvar:'tx.referer_request_uri=%{tx.1}'"
```

**Comment by dune73** [comment]
It's true that splitting ARGS and ARGS_NAMES is hard. But splitting URI and QS would be a good start.

**Comment by theMiddleBlue** [comment]
it's a big step forward, agree, But not for FP. Almost all matching occur in the querystring part of the referer sadly

**Comment by theMiddleBlue** [comment]
Adding to this issue other examples. Sorry if I'm not providing the raw data but it's really hard to extract and anonymize them because they come from a production env. Following a list of really common FP with high impact for UA and Referer request headers: 

## Request header User-Agent common FPs

#### Let's Encrypt ACME challenge bot blocked by 932200
transaction.messages.details.data: `Matched Data: /5.0 (compatible; let' found within REQUEST_HEADERS:User-Agent: mozilla/5.0 (compatible; let's encrypt validation server;  https://www.letsencrypt.org)`
transaction.messages.message: RCE Bypass Technique
transaction.messages.details.ruleId: 932200

#### DuckDuckGo search engine crawler blocked by 932200
transaction.messages.details.data: `Matched Data: 'duckduckbot-https/ found within REQUEST_HEADERS:User-Agent: 'duckduckbot-https/1.1; ( https://duckduckgo.com/duckduckbot)'`
transaction.messages.message: RCE Bypass Technique
transaction.messages.details.ruleId: 932200

#### Facebook for Android blocked by 932131
transaction.messages.details.data: `Matched Data: [fb_iab/fb4a fbav/403.0.0.27.81 ] found within REQUEST_HEADERS:user-agent: mozilla/5.0(linux android 12 sm-a528b build/sp1a.210812.016 wv) applewebkit/537.36(khtml like gecko) version/4.0 chrome/110.0.5481.154 mobile safari/537.36 [fb_iab/fb4a fbav/403.0.0.27.81 ]`
transaction.messages.message: Remote Command Execution: Unix Shell Expression Found
transaction.messages.details.ruleId: 932131


## Request header Referer common FPs

#### Browser highlight string functionality blocked by 942370
Example: [something like this](https://coreruleset.org/#:~:text=generic%20attack%20detection)
transaction.messages.details.data: `%20evidenze,'i found within REQUEST_HEADERS:referer: https://www.google.com/url?q=https://example.com/path/#:~:text=Inoltre%2C%20non%20vi%20sono%20evidenze,'immaginario%20comune%20(1).&sa=U&ved=2ahUKEwjSluGs8eX9AhWiSvEDHaaEChEQFnoECAgQBQ&usg=AOvVaw1_nmpM50Zh9-mGx1DsMw7j`
transaction.messages.message: Detects classic SQL injection probings 2/3
transaction.messages.details.ruleId: 942370

#### Something followed by the `?` character blocked by 932200
transaction.messages.details.data: `Matched Data: /sezione.jsp? found within REQUEST_HEADERS:Referer: https://example.com/sezione.jsp?id=17&lb=chi siamo&idx=1234`
transaction.messages.message: RCE Bypass Technique
transaction.messages.details.ruleId: 932200

**Comment by theseion** [comment]
I recently solved a similar issue by targeting the `Referer` header explicitly (932205, 932206). The difference to the original rule is the target and the regular expression that ensures that we are looking at an URL. Otherwise, if an attacker does not use an URL in the `Referer` header but just some exploit code, that would bypass detection (hence, two rules).

**Comment by theseion** [comment]
I've created a PR to fix the FPs in 932200 and created issues for the remaining two finds by @theMiddleBlue.

**Comment by dune73** [comment]
Thank you for picking this up.

**Comment by theseion** [comment]
Everything here appears to be solved (there's one open PR but it has a separate issue).
